src/Config/Parser/MetadataParser/MetadataParser.php

```diff
@@ -0,0 +1,956 @@
+        $fieldsFromMethods = self::getGraphQLTypeFieldsFromAnnotations($reflectionClass, $reflectionClass->getMethods(), Metadata\Field::class, $currentValue);
+
+        $typeConfiguration['fields'] = array_merge($fieldsFromProperties, $fieldsFromMethods);
+        $typeConfiguration = self::getDescriptionConfiguration($metadatas) + $typeConfiguration;
+
+        if (!empty($typeAnnotation->interfaces)) {
+            $typeConfiguration['interfaces'] = $typeAnnotation->interfaces;
+        } else {
+            $interfaces = array_keys(self::$map->searchClassesMapBy(function ($gqlType, $configuration) use ($reflectionClass) {
+                ['class' => $interfaceClassName] = $configuration;
+
+                $interfaceMetadata = self::getClassReflection($interfaceClassName);
+                if ($interfaceMetadata->isInterface() && $reflectionClass->implementsInterface($interfaceMetadata->getName())) {
+                    return true;
+                }
+
+                return $reflectionClass->isSubclassOf($interfaceClassName);
+            }, self::GQL_INTERFACE));
+
+            sort($interfaces);
+            $typeConfiguration['interfaces'] = $interfaces;
+        }
+
+        if (isset($typeAnnotation->resolveField)) {
+            $typeConfiguration['resolveField'] = self::formatExpression($typeAnnotation->resolveField);
+        }
+
+        $buildersAnnotations = array_merge(self::getMetadataMatching($metadatas, Metadata\FieldsBuilder::class), $typeAnnotation->builders);
+        if (!empty($buildersAnnotations)) {
+            $typeConfiguration['builders'] = array_map(function ($fieldsBuilderAnnotation) {
+                return ['builder' => $fieldsBuilderAnnotation->value, 'builderConfig' => $fieldsBuilderAnnotation->config];
+            }, $buildersAnnotations);
+        }
+
+        if (isset($typeAnnotation->isTypeOf)) {
+            $typeConfiguration['isTypeOf'] = $typeAnnotation->isTypeOf;
+        }
+
+        $publicMetadata = self::getFirstMetadataMatching($metadatas, Metadata\IsPublic::class);
+        if (null !== $publicMetadata) {
+            $typeConfiguration['fieldsDefaultPublic'] = self::formatExpression($publicMetadata->value);
+        }
+
+        $accessMetadata = self::getFirstMetadataMatching($metadatas, Metadata\Access::class);
+        if (null !== $accessMetadata) {
+            $typeConfiguration['fieldsDefaultAccess'] = self::formatExpression($accessMetadata->value);
+        }
+
+        return ['type' => $typeAnnotation->isRelay ? 'relay-mutation-payload' : 'object', 'config' => $typeConfiguration];
+    }
+
+    /**
+     * Create a GraphQL Interface type configuration from metadatas on properties.
+     */
+    private static function typeInterfaceMetadataToGQLConfiguration(ReflectionClass $reflectionClass, Metadata\TypeInterface $interfaceAnnotation): array
```

Comment:
Same here, this method always returns arrays with only 2 keys. You can describe the exact array shape and it would give PHPStan more information and make it's analysis more precision.

```suggestion
    /**
     * Create a GraphQL Interface type configuration from metadatas on properties.
     *
     * @return array{type: string, config: array}
     */
    private static function typeInterfaceMetadataToGQLConfiguration(ReflectionClass $reflectionClass, Metadata\TypeInterface $interfaceAnnotation): array
```

In this particular example you can even use a scalar type, since the value of the key 'type' is always 'interface' ([docs](https://phpstan.org/writing-php-code/phpdoc-types#literals-and-constants)):
```
/*
 * @return array{type: 'interface', config: array}
 */
```
But I don't know if scalar values can be mixed with array shapes syntax and moreover if it's necessary, because PHPStan might extract this information from the PHP's return statement directly.